lib/src/utils/uri_extension.dart

```diff
@@ -21,7 +21,64 @@ import 'dart:core';
 import 'package:matrix/src/client.dart';
 
 extension MxcUriExtension on Uri {
-  /// Returns a download Link to this content.
+  /// Transforms this `mxc://` Uri into a `http` resource, which can be used
+  /// to download the content.
+  ///
+  /// Throws an exception if the scheme is not `mxc` or the homeserver is not
+  /// set.
+  ///
+  /// Important! To use this link you have to set a http header like this:
+  /// `headers: {"authentication": "Bearer ${client.accessToken}"}`
+  Uri getAuthenticatedDownloadLink(Client client) {
+    if (!isScheme('mxc')) throw Exception('Uri has unknown scheme "mxc"');
+    final homeserver = client.homeserver;
+    if (homeserver == null) {
+      throw Exception(
+          'Homeserver must be specified to generate link from mxc uri');
+    }
+    return homeserver.resolve(
+        '_matrix/client/v1/media/download/$host${hasPort ? ':$port' : ''}$path');
+  }
+
+  /// Transforms this `mxc://` Uri into a `http` resource, which can be used
+  /// to download the content with the given `width` and
+  /// `height`. `method` can be `ThumbnailMethod.crop` or
+  /// `ThumbnailMethod.scale` and defaults to `ThumbnailMethod.scale`.
+  /// If `animated` (default false) is set to true, an animated thumbnail is requested
+  /// as per MSC2705. Thumbnails only animate if the media repository supports that.
+  ///
+  /// Throws an exception if the scheme is not `mxc` or the homeserver is not
+  /// set.
+  ///
+  /// Important! To use this link you have to set a http header like this:
+  /// `headers: {"authentication": "Bearer ${client.accessToken}"}`
+  Uri getAuthenticatedThumbnailLink(Client client,
+      {num? width,
+      num? height,
+      ThumbnailMethod? method = ThumbnailMethod.crop,
+      bool? animated = false}) {
+    if (!isScheme('mxc')) throw Exception('Uri has unknown scheme "mxc"');
+    final homeserver = client.homeserver;
+    if (homeserver == null) {
+      throw Exception(
+          'Homeserver must be specified to generate link from mxc uri');
+    }
+    return Uri(
+      scheme: homeserver.scheme,
+      host: homeserver.host,
+      path:
+          '/_matrix/client/v1/media/thumbnail/$host${hasPort ? ':$port' : ''}$path',
+      port: homeserver.port,
+      queryParameters: {
+        if (width != null) 'width': width.round().toString(),
+        if (height != null) 'height': height.round().toString(),
+        if (method != null) 'method': method.toString().split('.').last,
+        if (animated != null) 'animated': animated.toString(),
+      },
+    );
+  }
+
+  @Deprecated('Use `getAuthenticatedDownloadLink()` instead')
```

Comment:
Are these deprecated? The guide asks you to check the server supported version and then use the the endpoint corresponding to the supported version 

> If the server doesn’t support Matrix 1.11, the existing /_matrix/media/* endpoints should continue to be used.

https://matrix.org/blog/2024/06/20/matrix-v1.11-release/